py_lesson4_flask/main.py

```diff
@@ -0,0 +1,110 @@
+# Написать программу, которая скачивает изображения с заданных URL-адресов и сохраняет их на диск. Каждое изображение
+# должно сохраняться в отдельном файле, название которого соответствует названию изображения в URL-адресе.
+# Например, URL-адрес: https://example/images/image1.jpg -> файл на диске: image1.jpg
+# — Программа должна использовать многопоточный, многопроцессорный и асинхронный подходы.
+# — Программа должна иметь возможность задавать список URL-адресов через аргументы командной строки.
+# — Программа должна выводить в консоль информацию о времени скачивания каждого изображения и общем времени выполнения
+#   программы.
+
+import requests
+import os
+import time
+import argparse
+from multiprocessing.pool import ThreadPool
+from concurrent.futures import ThreadPoolExecutor
+import aiohttp
+import asyncio
+
+
+def download_image(url):
+    folder_save = 'data_homework'
+    if not os.path.exists(folder_save):
+        os.mkdir(folder_save)
+    try:
+        response = requests.get(url)
+        if response.status_code == 200:
+            filename = os.path.basename(url)
+            with open(f'{folder_save}/{filename}', 'wb') as file:
```

Comment:
Чтобы код работал на различных платформах лучше сделать так
```suggestion
            with open(os.path.join(folder_save, filename), 'wb') as file:
```